src/expint.jl

```diff
@@ -122,27 +124,45 @@ function En_cf_nogamma(x::Number, ν::Number, n::Int=1000)
     end
 end
 
+# Calculate Γ(1 - ν) * z^(ν-1) safely
+function En_safe_gamma_term(ν::Number, z::Number)
+    g = gamma(1 - ν)
+    p = float(z)^(ν - 1)
+    if abs(g) == Inf || abs(p) == Inf
+        return exp((ν - 1)*log(complex(z)) + loggamma(1 - ν))
+    else
+        if g == 0
+            return 0
+        else
+            return g*p
+        end
+    end
+end
 
-function En_cf_gamma(x::Number, ν::Number, n::Int=1000)
+# continued fraction for En(ν, z) that uses the gamma function:
+# https://functions.wolfram.com/GammaBetaErf/ExpIntegralE/10/0005/
+# returns the two terms from the above equation separately
+function En_cf_gamma(ν::Number, z::Number, n::Int=1000)
     A = float(1 - ν)
     B::typeof(A) = 1
     Bprev::typeof(A) = 0
     Aprev::typeof(A) = 1
+    ϵ = 10*eps(real(B))
     
     iters = 0
     j = 1
     for i = 2:n
         iters += 1
 
         A′ = A
-        term = iseven(i) ? (i - 1 - ν)*x : x
+        term = iseven(i) ? (i - 1 - ν)*z : z
```

Comment:
potentially not type-stable, e.g. if `z` is real and `ν` is complex.     Maybe just unroll the loop into `for i = 2:2:n` and do two iterations?